CHANGELOG.md

```diff
@@ -8,6 +8,8 @@ All notable changes to this project will be documented in this file.
   ([#823](https://github.com/open-telemetry/weaver/pull/823) by @lmolkova)
 - Don't serialize default values and empty arrays when resolving semantic conventions.
   ([#822](https://github.com/open-telemetry/weaver/pull/822) by @lmolkova)
+- Add support for a three registry dependency chain, a->b->c. This pattern is useful when making narrow application registries that depend on a corporate registry based on the OpenTelemetry semantic conventions.
```

Comment:
Is there a difference between supporting a>b->c and supporting arbitrary non-circular chain?

Reply:
Not really. I added in a max-depth constant after which it will throw an error as some protection. I set this to 3 for now but could make it much larger. The circular reference check doesn't rely on the limit. Three was my use-case,  should we make it larger?

Reply:
I had the same question :-)

Having a maximum limit seems like a good idea, but I think we can go up to 10 levels to cover the most extreme cases.

Reply:
Set this to 10